template/cordova/lib/PluginHandler.js

```diff
@@ -85,12 +85,12 @@ var handlers = {
             var src = obj.src;
             var dest = src;
             var type = obj.type;
+            var targetDir = obj.targetDir || '';
 
             if(type === 'projectReference') {
-                project.addProjectReference(path.join(plugin.dir,src), getTargetConditions(obj));
-            }
-            else {
-                var targetDir = obj.targetDir || '';
+                dest = path.join('plugins', plugin.id, targetDir, src);
+                project.addProjectReference(dest, getTargetConditions(obj));
```

Comment:
After looking more closely, there are two different calls being made here. One is to addProjectReference(), and the other is to addReference().